src/main/java/io/cdap/plugin/SCPRemotetoRemoteAction.java

```diff
@@ -0,0 +1,293 @@
+
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(SCPRemotetoRemoteAction.PLUGIN_NAME)
+@Description("This action will connect to a Bastion Host and execute a SCP command to copy a file from Host A to Host B.")
+public class SCPRemotetoRemoteAction extends Action {
+
+    public static final String PLUGIN_NAME = "SCPRemote";
+    private static final Logger LOG = LoggerFactory.getLogger(SCPRemotetoRemoteAction.class);
+
+    private final SCPRemotetoRemoteActionConfig config;
+
+    @VisibleForTesting
+    public SCPRemotetoRemoteAction(SCPRemotetoRemoteActionConfig config) {
+        this.config = config;
+    }
+
+    /**
+     * This function is executed by the Pipelines framework when the Pipeline is deployed. This
+     * is a good place to validate any configuration options the user has entered. If this throws
+     * an exception, the Pipeline will not be deployed and the user will be shown the error message.
+     */
+    @Override
+    public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+        super.configurePipeline(pipelineConfigurer);
+        LOG.debug(String.format("Running the 'configurePipeline' method of the %s plugin.", PLUGIN_NAME));
+        config.validate();
+    }
+
+    @Override
+    public void run(ActionContext context) throws JSchException, IOException {
+        LOG.debug(String.format("Running the 'run' method of the %s plugin.", PLUGIN_NAME));
+        config.validate();
+
+        byte[] key = config.getPrivateKey();
+        byte[] passphrase = config.getPassphrase();
+        JSch jsch = new JSch();
+        jsch.addIdentity("key", key, null,passphrase);
+        Session session = jsch.getSession(config.getUserNameBastion(),config.getHostBastion(),config.getPort());
+
+        session.setConfig("StrictHostKeyChecking", "no");
+        session.connect();
+        ChannelExec channel = (ChannelExec) session.openChannel("exec");
+
+        String userA = config.getUserNameA();
+        String hostA = config.getHostA();
+        String source = config.getSource();
+        String pathSource = userA+"@"+hostA+":"+source;
+
+        String userB = config.getUserNameB();
+        String hostB = config.getHostB();
+        String dest= config.getDest();
+        String pathDest = userB+"@"+hostB+":"+dest;
+
+        String dirFlag = config.getDirFlag();
+        String compressFlag = config.getCompressFlag();
+        String verboseFlag = config.getVerboseFlag();
+
+        //Host A -> Host B
+        channel.setCommand("scp "+compressFlag+ " "+verboseFlag+
+            " " +dirFlag +" " +pathSource +" " +pathDest);
+
+        StringBuilder outputBuffer = new StringBuilder();
+        StringBuilder errorBuffer = new StringBuilder();
+
+        InputStream in = channel.getInputStream();
+        InputStream err = channel.getExtInputStream();
+
+        channel.connect();
+
+        byte[] tmp = new byte[1024];
+        while (true) {
+            while (in.available() > 0) {
```

Comment:
Better not to use available. Just read until the end.

```
int len = in.read(tmp, 0, tmp.length);
while (len >= 0) {
  output.append(new String(tmp, 0, len, StandardCharsets.UTF_8));
  len = in.read(tmp, 0, tmp.length);
}
```

Reply:
Also, turn this into a private method to have cleaner code

Reply:
Also, always provide a Charset when converting `byte[]` to `String` and vice versa.